aten/src/ATen/native/cpu/ReduceOpsKernel.cpp

```diff
@@ -155,6 +155,16 @@ static void prod_kernel_impl(TensorIterator& iter) {
   });
 }
 
+static void nanprod_kernel_impl(TensorIterator& iter) {
```

Comment:
For simplicity what would you think of making this analogous to `prod_kernel_impl`? 

You're not implementing a vectorized CPU version in this PR, right?